core/src/main/java/com/linecorp/armeria/client/ClientRequestContextBuilder.java

```diff
@@ -103,29 +116,24 @@ public ClientRequestContextBuilder requestOptions(RequestOptions requestOptions)
      * Returns a new {@link ClientRequestContext} created with the properties of this builder.
      */
     public ClientRequestContext build() {
-        final Endpoint endpoint;
-        if (this.endpoint != null) {
-            endpoint = this.endpoint;
+        final EndpointGroup endpointGroup;
+        if (this.endpointGroup != null) {
+            endpointGroup = this.endpointGroup;
         } else {
-            endpoint = Endpoint.parse(authority());
+            endpointGroup = Endpoint.parse(authority());
         }
-
-        final CancellationScheduler responseCancellationScheduler;
+        final DefaultClientRequestContext ctx;
         if (timedOut()) {
-            responseCancellationScheduler = CancellationScheduler.finished(false);
+            ctx = getCtx(CancellationScheduler.finished(false));
         } else {
-            responseCancellationScheduler = CancellationScheduler.ofClient(0);
-            responseCancellationScheduler.initAndStart(eventLoop(), noopCancellationTask);
+            ctx = getCtx(CancellationScheduler.ofClient(0));
+            ctx.whenInitialized().handle((unused1, unused2) -> {
+                ctx.responseCancellationScheduler().initAndStart(eventLoop(), noopCancellationTask);
+                return null;
+            });
         }
 
-        final DefaultClientRequestContext ctx = new DefaultClientRequestContext(
-                eventLoop(), meterRegistry(), sessionProtocol(),
-                id(), method(), requestTarget(), options, request(), rpcRequest(),
-                requestOptions, responseCancellationScheduler,
-                isRequestStartTimeSet() ? requestStartTimeNanos() : System.nanoTime(),
-                isRequestStartTimeSet() ? requestStartTimeMicros() : SystemInfo.currentTimeMicros());
-
-        ctx.init(endpoint);
+        ctx.init(endpointGroup);
```

Comment:
`ctx.whenInitialized()` is completed by `ClientUtil`. As `ClientUtil` wasn't used to build `ClientRequestContext`, we need to complete it manually.
```java
ctx.init(endpointGroup).handle((unused, cause) -> {
    ctx.finishInitialization(cause == null);
    return null;
});
```